### PR TITLE
feat(deletion) Simplify deletion abort behavior

### PR DESCRIPTION
**Pull request by markstory**
Dan asked a really good question in my previous change about how deletions get aborted=True. Turns out that *never* happens.

I've dropped the `aborted` column from conditions and plan to remove it from the schema as well. In it's place deletion tasks can have implement a `should_proceed` hook to verify that the object has not been undeleted. I've implemented this hook for repositories as we don't actually have an API to undelete projects and teams.

I've removed the metric from repository pending_delete signal as it is in use.

### CODE REVIEW

File: src/sentry/deletions/base.py [hunk 1]
```diff
@@ -187,9 +199,9 @@ def chunk(self, num_shards=None, shard_id=None):
             if not queryset:
                 return False
 
-            self.delete_bulk(queryset)
+            has_more = self.delete_bulk(queryset)
             remaining -= query_limit
-        return True
```
Comment (markstory) [inline src/sentry/deletions/base.py:192]: Always returning true results in additional tasks being spawned for every delete we do. These tasks do no work but still consume resources. Now only if the last `delete_bulk` operation has more work will we spawn additional tasks.

File: src/sentry/tasks/deletion.py [hunk 2]
```diff
@@ -84,6 +82,18 @@ def run_deletion(deletion_id, first_pass=True):
...
+    if not task.should_proceed(instance):
+        logger.info(
+            "object.delete.aborted",
+            extra={
+                "object_id": deletion.object_id,
+                "transaction_id": deletion.guid,
+                "model": deletion.model_name,
+            },
+        )
+        deletion.delete()
```
Comment (markstory) [inline src/sentry/tasks/deletion.py:94]: No point in having deletion jobs with `aborted=true` in them when we can just remove those records.

File: src/sentry/deletions/base.py [hunk 3]
```diff
@@ -173,6 +181,10 @@ def chunk(self, num_shards=None, shard_id=None):
         """
         query_limit = self.query_limit
         remaining = self.chunk_size
+        has_more = True
+        # TODO(mark) This loop results in all of the work being done in
+        # one chunk/task. This loop would need to be removed if we want
+        # smaller chunks of work.
```
Comment (wedamija) [inline src/sentry/deletions/base.py:187]: I don't quite understand this comment. It looks like `remaining` is just `chunk_size`, so if we have more than `chunk_size` rows we'll exit the loop

**Review by wedamija — APPROVED** [review]

File: src/sentry/deletions/base.py (hunk 3 above)
Comment (markstory) [reply src/sentry/deletions/base.py]: You're right I overlooked the `or` coalescing in the `__init__`, and was further distracted by tasks that only delete a single row returning true when there was no additional work to be done.

**Pull request merged**